redmine_import_time_entry/models/redmine_backend.py

```diff
@@ -0,0 +1,119 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    This module copyright (C) 2015 - Present Savoir-faire Linux
+#    (<http://www.savoirfairelinux.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv import orm, fields
+from openerp.tools.translate import _
+from openerp.addons.connector_redmine.unit.import_synchronizer import (
+    import_batch)
+from openerp.addons.connector.session import ConnectorSession
+from openerp.tools import DEFAULT_SERVER_DATE_FORMAT, \
+    DEFAULT_SERVER_DATETIME_FORMAT
+
+from datetime import datetime, timedelta
+
+import logging
+_logger = logging.getLogger(__name__)
+
+
+class redmine_backend(orm.Model):
+    _inherit = 'redmine.backend'
+
+    _columns = {
+        'contract_ref': fields.char(
+            'Contract # field name',
+            help="The field in Redmine used to relate a project in Redmine "
+            "to a project in Odoo. Each redmine project must have a unique "
+            "value for this attribute."
+        ),
+        'time_entry_last_update': fields.datetime(
+            'Last Time Entry Update', required=True,
+        ),
+        'time_entry_import_activate': fields.boolean(
+            'Activate Time Entry Import'
+        ),
+        'time_entry_number_of_days': fields.integer(
+            'Time Entries - Number of days',
+            help="Number of days used when fetching the time entries.",
+            required=True
+        )
+    }
+
+    _defaults = {
+        'time_entry_import_activate': True,
+        'time_entry_number_of_days': 14,
+        'time_entry_last_update': datetime(1900, 1, 1).strftime(
+            DEFAULT_SERVER_DATETIME_FORMAT),
+    }
+
+    def check_contract_ref(self, cr, uid, ids, context=None):
+        """
+        Check if the contract_ref field exists in redmine
+        """
+        backend = self.browse(cr, uid, ids[0], context=context)
+        adapter = self._get_base_adapter(cr, uid, ids, context=context)
+
+        try:
+            adapter._auth()
+        except Exception:
+            raise orm.except_orm(
+                _('Error'), _('Could not connect to Redmine'))
```

Comment:
This would make debugging very difficult, how about:
```python
except Exception as e:
    raise orm.except_orm(type(e), _('Could not connect to Redmine: %s') % ustr(e))
```